src/app/parcels/ExpandedParcelDetails.tsx

```diff
@@ -11,20 +17,56 @@ const sortByTimestampWithMostRecentFirst = (events: EventTableRow[]): EventTable
     return events.sort((eventA, eventB) => eventB.timestamp.getTime() - eventA.timestamp.getTime());
 };
 
-const ExpandedParcelDetails = async ({ parcelId }: Props): Promise<React.ReactElement> => {
-    if (!parcelId) {
-        return <></>;
+function getErrorMessageForExpandedParcelDetailsError(
+    error: FetchExpandedParcelDetailsError
+): string {
+    switch (error.type) {
+        case "failedToFetchParcelDetails":
+            return `Failed to fetch parcel details. Log ID: ${error.logId}`;
+        case "clientDetailDoesNotExist":
+            return `Client detail cannot be found. Log ID: ${error.logId}`;
     }
-    const expandedParcelDetails = await getExpandedParcelDetails(parcelId);
+}
+
+const ExpandedParcelDetailsView = ({ parcelId }: Props): ReactElement => {
+    const [parcelDetails, setParcelDetails] = useState<ExpandedParcelDetails | null>(null);
+    const [errorMessage, setErrorMessage] = useState<string | null>(null);
+
+    const fetchAndSetParcelDetails = useCallback(async (): Promise<void> => {
+        if (!parcelId) {
+            return;
+        }
+
+        const { parcelDetails: expandedParcelDetails, error } =
+            await getExpandedParcelDetails(parcelId);
+
+        if (error) {
+            const newErrorMessage = getErrorMessageForExpandedParcelDetailsError(error);
+            setErrorMessage(newErrorMessage);
+            return;
+        }
+
+        setParcelDetails(expandedParcelDetails);
+    }, [parcelId]);
+
+    useEffect(() => {
+        void fetchAndSetParcelDetails();
+    }, [fetchAndSetParcelDetails]);
```

Comment:
just for my own educational benefit, can you please explain to me why we would want to do this useCallback thing?

Reply:
The main reason is to make the useEffect simple and easy to read. If we instead had the content of `fetchAndSetParcelDetails` inside the use effect, then future devs would have to read the content to understand what happens in the use effect.

Reply:
Thought so, nice, thank you!